### PR TITLE
ENH: distutils: add compiler classes for Intel compilers + MSVC-built Python

### PR DESCRIPTION
**Pull request by rgommers**
Thanks to Intel for contributing this patch.

### CODE REVIEW

**Comment by njsmith** [comment]
Oo, with itanium support even.

File: numpy/distutils/ccompiler.py [hunk 1]
```diff
@@ -509,7 +509,10 @@ def CCompiler_cxx_compiler(self):
         The C++ compiler, as a `CCompiler` instance.
 
     """
-    if self.compiler_type=='msvc': return self
+    if (self.compiler_type == 'msvc' or self.compiler_type == 'intelw' or
```
Comment (matthew-brett) [inline numpy/distutils/ccompiler.py]: Maybe

```
if self.compiler_type in ('msvc', 'intelw', 'intelemw'):
```

Comment (rgommers) [inline numpy/distutils/ccompiler.py]: indeed. done.

**Comment by charris** [comment]
Thanks Ralf. Failed test was due to waf.io being offline.

**Comment by rgommers** [comment]
I'll have a look at the Waf situation

**Pull request merged**